### DISCUSSION TITLE
Run syncoid with --delete-target-snapshots and delete target snapshots first?

### DISCUSSION THREAD

**Discussion by numerfolt** [jimsalterjrs/sanoid#960] [Q&A]
I currently want to backup my 2TB main SSD with ZFS to a pair of 2TB HDDs in external enclosures.
But even though I tell syncoid to delete target snapshots that are not on the source, my backup pool is much more full than my main pool.

This leads to me not being able to backup to the HDD anymore, because it doesnt have enough space on it.
How can I avoid this?

My command is:
`syncoid -r --no-sync-snap --create-bookmark --delete-target-snapshots main-pool backup-pool/main-pool`

The only option I found in 1 hour of websearch is this script: https://gitlab.com/KodyVB/zfs-syncer

**Comment by phreaker0** [comment]
There can be multiple reasons why it it needs more space on the target pool. The main ones which come to mind are recordsize and compression. If you use a better compression or a larger recordsize on the source it will take up less space.

So if the target is only uses as backup you can choose a slower but better compression option and/or a larger recordsize.
